AnalysisStep/plugins/ZZCandidateFiller.cc

```diff
@@ -443,528 +440,755 @@ void ZZCandidateFiller::produce(edm::Event& iEvent, const edm::EventSetup& iSetu
 
     //--- store good isolated leptons that are not involved in the current ZZ candidate
     int nExtraLep = 0;
-    math::XYZTLorentzVector lepForWH; 
-    int idOfLepForWH = 0;
-    float extraLepPtMax = 0.;
-    for( vector<reco::CandidatePtr>::const_iterator lepPtr = goodisoleptonPtrs.begin(); lepPtr != goodisoleptonPtrs.end(); ++ lepPtr ) {
+    SimpleParticleCollection_t associatedLeptons;
+    for (vector<reco::CandidatePtr>::const_iterator lepPtr = goodisoleptonPtrs.begin(); lepPtr != goodisoleptonPtrs.end(); ++lepPtr){
       const reco::Candidate* lep = lepPtr->get();
-      if( reco::deltaR( lep->p4(), Z1L1->p4() ) > 0.02 &&
-	  reco::deltaR( lep->p4(), Z1L2->p4() ) > 0.02 &&
-	  reco::deltaR( lep->p4(), Z2L1->p4() ) > 0.02 &&
-	  reco::deltaR( lep->p4(), Z2L2->p4() ) > 0.02 ){
-	nExtraLep++;
-	myCand.addUserCand("ExtraLep"+to_string(nExtraLep),*lepPtr);
-	if(lep->pt()>extraLepPtMax){
-	  lepForWH = lep->p4();
-	  idOfLepForWH = lep->pdgId();
-	  extraLepPtMax = lep->pt();
-	}
+      if (
+        reco::deltaR(lep->p4(), Z1L1->p4()) > 0.02 &&
+        reco::deltaR(lep->p4(), Z1L2->p4()) > 0.02 &&
+        reco::deltaR(lep->p4(), Z2L1->p4()) > 0.02 &&
+        reco::deltaR(lep->p4(), Z2L2->p4()) > 0.02
+        ){
+        myCand.addUserCand("ExtraLep"+to_string(nExtraLep), *lepPtr);
```

Comment:
The fact that nExtraLep++ has been moved down means that the labelling of UserCand changes (will start from 0 instead than from 1). I think this is not intended - code elsewhere assume the labelling starts from 1. See eg HZZ4lNtupleMaker.cc:1664. I would move back the increment where it was, to preserve the previous behaviuor, unless this is intended (but all existing code should be adapted then).